Annotation is a key step in genomic analysis, where meaningful attributes or properties are assigned to a set of variants under study. For example, in a genetic study of case and control groups, variants can be annotated with predictions of their pathogenicity and conservation scores at the variant loci. We developed a robust annotation infrastructure within our [Genotypes and Phenotypes in Families](https://gpf.sfari.org/) system.

Users customize the annotation process by specifying the genomic properties of interest in an annotation pipeline, which is represented as a YAML file. These properties are sourced from one or more Genomic Resource Repositories (GRR). We maintain a public [GRR](https://grr.iossifovlab.com/) that includes resources for population frequencies (e.g. gnomAD), conservation scores (e.g. CADD, phyloP, phastCons), and missense scores (e.g. MPC). Users need only specify the resource identifiers they wish to use for annotating their variants. The flexibility of GPFâ€™s GRR infrastructure allows users to work with both public and private resources in their custom workflows.

mini-GRR is a repository containing a set of mock genomic resources designed to demonstrate the capabilities of GRR and assist users in building their private GRRs. These resources cover various file types (e.g., tsv, bedgraph, tabix, vcf, bw), score types (e.g., position, allele, gene), reference types (0-based or 1-based), and region types (single nucleotide or region). 

Further documentation on using GRR is available at [https://iossifovlab.com/gpfuserdocs/administration/genomic_resources.html]( https://iossifovlab.com/gpfuserdocs/administration/genomic_resources.html). Additional information about our annotation infrastructure can be found at [https://iossifovlab.com/gpfuserdocs/administration/annotation.html]( https://iossifovlab.com/gpfuserdocs/administration/annotation.html).
